source/core/shape.cc

```diff
@@ -0,0 +1,428 @@
+/* ---------------------------------------------------------------------
+ *
+ * Copyright (C) 2019 - by the Lethe authors
+ *
+ * This file is part of the Lethe library
+ *
+ * The Lethe library is free software; you can use it, redistribute
+ * it, and/or modify it under the terms of the GNU Lesser General
+ * Public License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ * The full text of the license can be found in the file LICENSE at
+ * the top level of the Lethe distribution.
+ *
+ * ---------------------------------------------------------------------
+ */
+#include <core/shape.h>
+
+template <int dim>
+double
+Shape<dim>::displaced_volume(const double fluid_density)
+{
+  StandardExceptions::ExcNotImplemented();
```

Comment:
```suggestion
  StandardExceptions::ExcNotImplemented();
```
You need to throw this one no?
Why not make this a pure virtual function?